pkg/migration/controllers/migration.go

```diff
@@ -730,159 +766,29 @@ func (m *MigrationController) migrateResources(migration *stork_api.Migration) e
-		return nil, err
-	}
-	allObjects := make([]runtime.Unstructured, 0)
-	resourceInfos := make([]*stork_api.ResourceInfo, 0)
-
-	for _, group := range m.discoveryHelper.Resources() {
-		groupVersion, err := schema.ParseGroupVersion(group.GroupVersion)
-		if err != nil {
-			return nil, err
-		}
-		if groupVersion.Group == "extensions" {
-			continue
-		}
-
-		resourceMap := make(map[types.UID]bool)
-		for _, resource := range group.APIResources {
-			if !resourceToBeMigrated(migration, resource) {
-				continue
-			}
-
-			for _, ns := range migration.Spec.Namespaces {
-				var dynamicClient dynamic.ResourceInterface
-				if !resource.Namespaced {
-					dynamicClient = m.dynamicInterface.Resource(groupVersion.WithResource(resource.Name))
-				} else {
-					dynamicClient = m.dynamicInterface.Resource(groupVersion.WithResource(resource.Name)).Namespace(ns)
-				}
-
-				var selectors string
-				// PVs don't get the labels from their PVCs, so don't use
-				// the label selector
-				if resource.Kind != "PersistentVolume" {
-					selectors = labels.Set(migration.Spec.Selectors).String()
-				}
-				objectsList, err := dynamicClient.List(metav1.ListOptions{
-					LabelSelector: selectors,
-				})
-				if err != nil {
-					return nil, err
-				}
-				objects, err := meta.ExtractList(objectsList)
-				if err != nil {
-					return nil, err
-				}
-				for _, o := range objects {
-					runtimeObject, ok := o.(runtime.Unstructured)
-					if !ok {
-						return nil, fmt.Errorf("Error casting object: %v", o)
-					}
-
-					migrate, err := m.objectToBeMigrated(migration, resourceMap, runtimeObject, ns)
-					if err != nil {
-						return nil, fmt.Errorf("Error processing object %v: %v", runtimeObject, err)
-					}
-					if !migrate {
-						continue
-					}
-					metadata, err := meta.Accessor(runtimeObject)
-					if err != nil {
-						return nil, err
-					}
-					resourceInfo := &stork_api.ResourceInfo{
-						Name:      metadata.GetName(),
-						Namespace: metadata.GetNamespace(),
-						Status:    stork_api.MigrationStatusInProgress,
-					}
-					resourceInfo.Kind = resource.Kind
-					resourceInfo.Group = groupVersion.Group
-					// core Group doesn't have a name, so override it
-					if resourceInfo.Group == "" {
-						resourceInfo.Group = "core"
-					}
-					resourceInfo.Version = groupVersion.Version
-					resourceInfos = append(resourceInfos, resourceInfo)
-					allObjects = append(allObjects, runtimeObject)
-					resourceMap[metadata.GetUID()] = true
-				}
-			}
-		}
-		migration.Status.Resources = resourceInfos
-		err = sdk.Update(migration)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	return allObjects, nil
-}
-
 func (m *MigrationController) prepareResources(
 	migration *stork_api.Migration,
 	objects []runtime.Unstructured,
 ) error {
 	for _, o := range objects {
-		content := o.UnstructuredContent()
-		// Status shouldn't be migrated between clusters
-		delete(content, "status")
+		metadata, err := meta.Accessor(o)
```

Comment:
Whats the difference between delete(content, "status") and meta,Accessor(o) ? How does a call to Accessor(o) removes the need to have a delete? Some code comment here, to understand what meta.Accessor(o) does and why it eliminated the need to have an explicit delete call.